views.py

```diff
@@ -93,6 +98,43 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
                 bag_modified = 'true'
         metadata_dirty = istorage.getAVU(res_root, 'metadata_dirty')
 
+
+
+    if is_zip_download:
+
+        if not path.endswith(".zip"): # requesting folder that needs to be zipped
+            input_path = path.split(res_id)[1]
+            random_hash = random.getrandbits(16)
+            random_hash_path = 'zips/{res_id}/{rand_folder}'.format(res_id=res_id, rand_folder=random_hash)
+            output_path = '{random_hash_path}{path}.zip'.format(random_hash_path=random_hash_path, path=input_path)
+
+            if use_async:
+                task = create_temp_zip.apply_async((res_id, input_path, output_path), countdown=3)
+                delete_zip.apply_async((res_id, random_hash_path), countdown=(20 * 60))  # delete after 20 minutes
+                download_path = request.path.split("zips")[0] + output_path
```

Comment:
Wouldn't it always be ```zips``` in ```request.path.split("zips")[0]```?

Reply:
no, it would contain the string before zips... so the path before zips.  "zips" string is omitted from all results in the split.

Reply:
OK. got it now.